packages/eslint-plugin-react-hooks/__tests__/ESLintRulesOfHooks-test.js

```diff
@@ -971,6 +1001,70 @@ const tests = {
       `,
       errors: [classError('useState')],
     },
+    {
+      code: `
+        function MyComponent({ theme }) {
+          const onClick = useEvent(() => {
+            showNotification(theme);
+          });
+          return <Child onClick={onClick}></Child>;
+        }
+      `,
+      errors: [useEventError()],
+    },
+    {
+      code: `
+        function MyComponent({ theme }) {
+          const onClick = useEvent(() => {
+            showNotification(theme);
+          });
+          let jsx;
+          if (theme === 'dark') {
+            jsx = <Child onClick={onClick} className="dark" />
+          } else {
+            jsx = <Child onClick={onClick} className="light" />
+          }
+          const jsx2 = theme === 'dark'
+            ? <Child onClick={onClick} className="dark" />
+            : <Child onClick={onClick} className="light" />;
+
+          return jsx;
+        }
+      `,
+      errors: [
+        useEventError(),
+        useEventError(),
+        useEventError(),
+        useEventError(),
+      ],
+    },
+    {
+      code: `
+        function MyComponent({ theme }) {
+          const onClick = useEvent(() => {
+            showNotification(theme);
+          });
+          return <Child onClick={onClick.bind(null)}></Child>;
+        }
+      `,
+      errors: [useEventError()],
+    },
+    {
+      code: `
+        function MyComponent({ theme }) {
+          const onClick = useEvent(() => {
+            showNotification(theme);
+          });
+          useMyHook(() => {
+            onClick();
+          });
+          const customOnClick = useMyHook(() => {
+            onClick();
+          });
+        }
```

Comment:
Why is this an error?

Reply:
I think this was my misunderstanding from our bikeshedding chats that useEvent should only be paired with useEffect. In practice iiuc it should ideally be used together but doesn't need to be enforced.